README.md
# physical2logical

Convert CSS physical properties to logical.

Converts margins, padding, and borders to logical values, allowing RTL and vertical languages to show correctly.

## Install

```commandline
pip install physical2logical
```

## Usage

```commandline
usage: css2logical [-h] [-r | --recursive | --no-recursive] [-u | --update | --no-update] [-a | --analyze | --no-analyze] [-f FILENAME]  [-l LOG] [-p PATTERN] source

Convert CSS physical properties to logical

positional arguments:
  source                Path to source directory or one file. Required

optional arguments:
  -h, --help            show this help message and exit
  -r, --recursive, --no-recursive
                        Recursive process of all files from source directory. (default: True)
  -u, --update, --no-update
                        Update files. (default: False)
  -a, --analyze, --no-analyze
                        Dry Run - Create a report with all changes (without actual source file changes). (default: False)
  -f FILENAME, --filename FILENAME
                        Path to file where to save reports (used only with -a | --analyze flag). (default: "./report.html")
  -l LOG, --log LOG     Log Level: All: 0, Debug: 10, Info: 20, Warning: 30, Error: 40 (default: 0)
  -p PATTERN, --pattern PATTERN
                        Comma separated list of glob patterns (default: '*.scss, *.css')


```

Based on https://gist.github.com/nyurik/d438cb56a9059a0660ce4176ef94576f

## Report

![report.png](report.png)

## Example

```commandline
css2logical -u ./src/scss/
```

![logs.png](logs.png)

![physical2logical-all-files.png](physical2logical-all-files.png)


CHANGELOG.md
<!--next-version-placeholder-->

## v2.3.1 (2023-01-21)
### Fix
* Fix default pattern list ([`deaeb5d`](https://github.com/easingthemes/physical2logical/commit/deaeb5df60e8b43843216a456eb9a1baf83f5e97))

## v2.3.0 (2023-01-21)
### Feature
* Add pattern as argument ([`be3a007`](https://github.com/easingthemes/physical2logical/commit/be3a0074815bdc79d7af45c2793628b3509ee595))

## v2.2.12 (2023-01-21)
### Fix
* Add log level to Readme ([`eb646bc`](https://github.com/easingthemes/physical2logical/commit/eb646bc05db7a87d4b674ebd5e7c714589be71bb))

## v2.2.11 (2023-01-21)
### Fix
* Change default log level ([`a0c6d23`](https://github.com/easingthemes/physical2logical/commit/a0c6d23b99516fd08505593fb6577296d5de020f))
* Change default log level ([`a6a55eb`](https://github.com/easingthemes/physical2logical/commit/a6a55eb8f9abb7c4010559c093569be1350bf2bc))

## v2.2.10 (2023-01-21)
### Fix
* Add log level as a param ([`1590426`](https://github.com/easingthemes/physical2logical/commit/159042693a3b3861757a0decada5174974765e95))

## v2.2.9 (2023-01-21)
### Fix
* Add more debug logs ([`3c9d829`](https://github.com/easingthemes/physical2logical/commit/3c9d829fd65b9482dadbc3c8945118421a6c76c4))

## v2.2.8 (2023-01-21)
### Fix
* Add debug logs ([`fb37649`](https://github.com/easingthemes/physical2logical/commit/fb376495723765c0172cc65c1d85fc3351d96006))

## v2.2.7 (2023-01-21)
### Fix
* Add report example image ([`95f12ef`](https://github.com/easingthemes/physical2logical/commit/95f12efcb34040f3e23c4adf1051ae4edc0ffcb4))

## v2.2.6 (2023-01-20)
### Fix
* Add relative default report file path ([`3970f5e`](https://github.com/easingthemes/physical2logical/commit/3970f5e4e292185b0a4988278c62a812b189ca5c))

## v2.2.5 (2023-01-20)
### Fix
* Log args on error ([`ee31c24`](https://github.com/easingthemes/physical2logical/commit/ee31c24b26e341119c170a15319738f830934a68))

## v2.2.4 (2023-01-20)
### Fix
* Filename validation fix ([`782264d`](https://github.com/easingthemes/physical2logical/commit/782264d329d49f3b815108ba15aefb02dfc2d54b))

## v2.2.3 (2023-01-20)
### Fix
* Add workflow commit changes ([`bb18c53`](https://github.com/easingthemes/physical2logical/commit/bb18c53eb60655030987d9784e1a6c751dcdf70e))


physical2logical/arguments.py
import argparse
import os
import sys
from pathlib import Path

from physical2logical.logger import logger


def log_and_exit(msg, callback):
    logger.error(f"{msg}")
    logger.empty()
    callback()
    sys.exit(1)


def is_valid_file_path(file_path):
    if os.path.exists(file_path):
        return True
    elif os.access(os.path.dirname(file_path), os.W_OK):
        return True
    else:
        return False


# can not write there
def get_args():
    parser = argparse.ArgumentParser(description='Convert CSS physical properties to logical')
    parser.add_argument('source',
                        help='Path to source directory or one file. Required')
    parser.add_argument('-r', '--recursive',
                        action=argparse.BooleanOptionalAction,
                        default=True,
                        help='Recursive process of all files from source directory.')
    parser.add_argument('-u', '--update',
                        action=argparse.BooleanOptionalAction,
                        default=False,
                        help='Update files.')
    parser.add_argument('-a', '--analyze',
                        action=argparse.BooleanOptionalAction,
                        default=False,
                        help='Dry Run - Create a report with all changes (without actual source file changes).')
    parser.add_argument('-f', '--filename',
                        default='./report.html',
                        help='Path to file where to save reports (used only with -a/--analyze flag).'
                             ' (default: "./report.html")')
    parser.add_argument('-l', '--log',
                        default=20,
                        type=int,
                        help='Log Level: All: 0, Debug: 10, Info: 20, Warning: 30, Error: 40'
                             ' (default: 20)')
    parser.add_argument('-p', '--pattern',
                        default="*.scss, *.css",
                        help="Comma separated list of glob patterns"
                             " (default: '*.scss, *.css')")

    if len(sys.argv) < 2:
        return log_and_exit(f"Path to source directory or one file is required", parser)

    args = parser.parse_args()

    root_path = Path(args.source)
    result_file = args.filename or "./report.html"
    is_recursive = args.recursive
    is_analyze = args.analyze
    is_update = args.update
    log_level = args.log
    pattern = args.pattern

    logger.setLevel(log_level)

    def print_args():
        logger.info_(f"Params: \n"
                     f"\tsource: {root_path}\n"
                     f"\tpattern: {pattern}\n"
                     f"\trecursive: {is_recursive}\n"
                     f"\tupdate: {is_update}\n"
                     f"\tanalyze: {is_analyze}\n"
                     f"\tresults file: {result_file}"
                     f"\tlog level: {log_level}")
        logger.empty()

    def callback():
        print_args()
        parser.print_help()

    if not is_analyze and not is_update:
        return log_and_exit("Action missing: [-a | --analyze] or [-u | --update]", callback)

    if not args.filename:
        logger.warning(f"Report filename not defined,using default '{result_file}'")
        logger.empty()

    is_valid_result_file = is_valid_file_path(result_file)
    if not is_valid_result_file:
        return log_and_exit(f"Filename must be valid file path. Received: {result_file}", callback)

    print_args()

    return is_analyze, is_update, is_recursive, root_path, result_file, pattern
